library/blivet.py

```diff
@@ -0,0 +1,709 @@
+
+
+class BlivetPartitionVolume(BlivetVolume):
+    def _type_check(self):
+        return self._device.type == 'partition'
+
+    def _get_device_id(self):
+        return self._blivet_pool._disks[0].name + '1'
+
+    def _create(self):
+        if self._device:
+            return
+
+        if self._blivet_pool:
+            parent = self._blivet_pool._device
+        else:
+            parent = self._blivet.devicetree.resolve_device(self._volume['pool'])
+
+        if parent is None:
+            # FAIL: failed to find pool
+            raise BlivetAnsibleError("failed to find pool '%s' for volume '%s'" % (self._blivet_pool['name'], self._volume['name']))
+
+        size = Size("256 MiB")
+        try:
+            device = self._blivet.new_partition(parents=[parent], size=size, grow=True, fmt=self._get_format())
+        except Exception:
+            # FAIL: failed to instantiate volume device
+            raise BlivetAnsibleError("failed set up volume '%s'" % self._volume['name'])
+
+        self._blivet.create_device(device)
+        try:
+            do_partitioning(self._blivet)
+        except Exception:
+            # FAIL: partition allocation failed: not enough space?
+            raise BlivetAnsibleError("partition allocation failed for volume '%s'" % self._volume['name'])
+
+        self._device = device
+
+
+class BlivetLVMVolume(BlivetVolume):
+    def _get_device_id(self):
+        return "%s-%s" % (self._blivet_pool._device.name, self._volume['name'])
+
+    def _create(self):
+        if self._device:
+            return
+
+        parent = self._blivet_pool._device
+        if parent is None:
+            # FAIL: failed to find pool
+            raise BlivetAnsibleError("failed to find pool '%s' for volume '%s'" % (self._blivet_pool['name'], self._volume['name']))
+
+        try:
+            size = Size(self._volume['size'])
+        except Exception:
+            # FAIL: invalid size specification
+            raise BlivetAnsibleError("invalid size '%s' specified for volume '%s'" % (self._volume['size'], self._volume['name']))
+
+        fmt = self._get_format()
+        if size > parent.free_space:
+            # FAIL: volume size greater than pool free space
+            raise BlivetAnsibleError("specified size for volume '%s' exceeds available space in pool '%s' (%s)" % (size,
+                                                                                                                   parent.name,
+                                                                                                                   parent.free_space))
+
+        try:
+            device = self._blivet.new_lv(name=self._volume['name'],
+                                         parents=[parent], size=size, fmt=fmt)
+        except Exception:
+            # FAIL: failed to create volume
+            raise BlivetAnsibleError("failed to set up volume '%s'" % self._volume['name'])
+
+        self._blivet.create_device(device)
```

Comment:
this throws a traceback:
`blivet3.errors.DependencyError: device type lvmlv requires unavailable_dependencies: dm`

Reply:
That would be in `libblockdev-dm`, but this package is not available in RHEL 7.

Reply:
@vojtechtrefny Is libblockdev-dm not available in RHEL 7 for some reason?

Reply:
@dwlehman @vojtechtrefny see my correction above, and sorry for the initial confusion - it is available, in Optional.

Reply:
fix in #33 - https://github.com/linux-system-roles/storage/pull/33/commits/3162351591c9d96c5f74c38c5c89adcdd796c4e7